### PR TITLE
chore(nextjs): Drop deprecations

### PR DESCRIPTION
**Pull request by dimkl**
## Description

Drop deprecations. Migration steps:
- use `NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY` instead of `NEXT_PUBLIC_CLERK_FRONTEND_API` env variable
- use `NEXT_PUBLIC_CLERK_JS_VERSION` instead of `CLERK_JS_VERSION`
- use `CLERK_SECRET_KEY` instead of `CLERK_API_KEY`
- use `publishableKey` instead of `frontendApi`
- use `isEmailLinkError` instead of `isMagicLinkError`
- use `EmailLinkErrorCode` instead of `MagicLinkErrorCode`
- use `useEmailLink` instead of `useMagicLink`
- use `@clerk/nextjs` instead of `@clerk/nextjs/app-beta`
- use `@clerk/nextjs` instead of `@clerk/nextjs/ssr`
- use `@clerk/nextjs` instead of `@clerk/nextjs/edge-middleware` or `@clerk/nextjs/edge-middlewarefiles`
- use middleware with `authMiddleware` instead of `withClerkMiddleware`
- avoid using exported constants: `API_URL`, `API_VERSION`, `CLERK_JS_URL`, `CLERK_JS_VERSION`, `DOMAIN`, `IS_SATELLITE`, `PROXY_URL`, `PUBLISHABLE_KEY`, `SECRET_KEY`, `SIGN_IN_URL`, `SIGN_UP_URL`

Extra:
- dropped `@clerk/clerk-sdk-node` dependency as there is no use anymore

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [x] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: .changeset/cyan-stingrays-own.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+'@clerk/nextjs': major
+---
+
+Drop deprecations. Migration steps:
```
Comment (anagstef) [inline .changeset/cyan-stingrays-own.md:5]: @dimkl  Should we mark these as breaking?

**Review by anagstef — APPROVED** [review]
💯

File: .changeset/cyan-stingrays-own.md (hunk 1 above)
Comment (dimkl) [reply .changeset/cyan-stingrays-own.md:5]: I thought that the major version bump should indicate that but we can also mark them as breaking changes. All the changes with `major` version are breaking.

Comment (dimkl) [reply .changeset/cyan-stingrays-own.md:5]: Merging this to continue with the cleanup and i will update the changeset if we want to somehow mark the changes as breaking.

**Pull request merged**